cartridges/int_alma/cartridge/scripts/helpers/almaHelpers.js

```diff
@@ -159,6 +159,95 @@ function haveExcludedCategory(productIds) {
 
     return haveExcludedCategoryReturn;
 }
+/**
+ * Get the full url for a page
+ * @param {string} pageName name of page for url
+ * @param {string} pageTemplate template of page for url
+ * @param {string} locale locale
+ * @returns {string} url
+ */
+function getFullPageUrl(pageName, pageTemplate, locale) {
+    var hostname = Site.getCurrent().getHttpsHostName();
+    var siteName = Site.getCurrent().getName();
+
+    return 'https://' + hostname + '/s/' + siteName + '/' + pageName + '/' + pageTemplate + '.html?lang=' + locale;
+}
+
+/**
+ * get orders items for website customer details
+ * @param {Object} order order
+ * @param {string} locale locale
+ * @returns {Object} items
+ */
+function getOrdersItemsForWebsiteCustomerDetails(order, locale) {
+    var forOf = require('*/cartridge/scripts/helpers/almaUtilsHelper').forOf;
+    var items = [];
+    forOf(order.getAllProductLineItems(), function (productLine) {
+        var product = productLine.getProduct();
+        var categories = [];
+        var fullPageUrl = '';
+        var productsCategories = '';
+
+        if (product.isMaster()) {
+            fullPageUrl = getFullPageUrl(product.getPageURL(), product.getID(), locale);
+            productsCategories = product.getAllCategories().toArray();
+        } else {
+            fullPageUrl = getFullPageUrl(product.getPageURL(), product.getMasterProduct().getID(), locale);
+            productsCategories = product.getMasterProduct().getAllCategories().toArray();
+        }
+
+        productsCategories.forEach(function (category) {
+            if (!categories.includes(category.getID())) {
+                categories.push(category.getID());
+            }
+        });
+        var item = {
```

Comment:
create a function buildItem something like that, try to reduce the complexity of your functions and create small dedicated functions

Reply:
Done